public/app/features/explore/state/query.ts

```diff
@@ -273,22 +294,44 @@ export const importQueries = (
     }
 
     let importedQueries = queries;
-    // Check if queries can be imported from previously selected datasource
-    if (sourceDataSource.meta?.id === targetDataSource.meta?.id) {
-      // Keep same queries if same type of datasource, but delete datasource query property to prevent mismatch of new and old data source instance
-      importedQueries = queries.map(({ datasource, ...query }) => query);
-    } else if (hasQueryExportSupport(sourceDataSource) && hasQueryImportSupport(targetDataSource)) {
-      const abstractQueries = await sourceDataSource.exportToAbstractQueries(queries);
-      importedQueries = await targetDataSource.importFromAbstractQueries(abstractQueries);
-    } else if (targetDataSource.importQueries) {
-      // Datasource-specific importers
-      importedQueries = await targetDataSource.importQueries(queries, sourceDataSource);
+    // If going to mixed, keep queries with source datasource
+    if (targetDataSource.name === MIXED_DATASOURCE_NAME) {
```

Comment:
Should we use `uid` instead of `name`? It's used in other places like `PublicDashboardDataSource.isMixedDatasource`. Maybe it'd make sense to have a util function exported to check if a data source is mixed?

Reply:
Well, it gets a little more complicated, some places use `datasource.meta.mixed` as well. I am going to change the equality to check for UID - I personally think this doesn't require a utility function but we can revisit it later.